noir-projects/noir-protocol-circuits/crates/types/src/address/aztec_address.nr

```diff
@@ -52,16 +66,55 @@ impl Deserialize<AZTEC_ADDRESS_LENGTH> for AztecAddress {
     }
 }
 
+impl ToPoint for AztecAddress {
+    fn to_point(self) -> Point {
+        // Calculate y^2 = x^3 - 17
+        let y_squared = pow(self.inner, 3) - 17;
+
+        // TODO (#8970): Failure validation
+        // We can see if y is square first, or we can soft fail with just sqrt(y_squared);
+        // If y is not square, the x-coordinate is not on the curve
+        // Do we throw here or can we soft continue and this is fine ? Test this with points not on curve to see what happens.
+        // let y_is_square = is_square(y_squared);
+        // assert(y_is_square);
```

Comment:
I found these comments a bit confusing, since it's not clear why the code is commented out - it seems like it was maybe forgotten about.

Given we're not handling any of this now, I'd just remove all of this.
```suggestion
        // TODO (#8970): Handle cases where we cannot recover a point from an address
```